### PR TITLE
Enables the SEO settings tab for Jetpack sites.

### PR DESCRIPTION
**Pull request by roundhill**
Run this branch alongside https://github.com/Automattic/jetpack/pull/5307 to get settings saving from calypso to the Jetpack site.

cc @vindl

### CODE REVIEW

File: client/my-sites/site-settings/seo-settings/controller.js [hunk 1]
```diff
@@ -34,8 +34,8 @@ export default {
 			return;
 		}
 
-		// redirect seo and analytics tabs to general for Jetpack sites
-		if ( site.jetpack ) {
+		// redirect analytics tab to general for Jetpack sites
+		if ( site.jetpack && basePath === '/settings/analytics' ) {
```
Comment (mtias) [inline client/my-sites/site-settings/seo-settings/controller.js]: When would path be `/settings/analytics` here?

**Comment by roundhill** [comment]
ToDo:

- [ ] Add check for the Jetpack version that Adv. SEO will ship with

File: client/my-sites/site-settings/seo-settings/form.jsx [hunk 2]
```diff
@@ -95,6 +104,12 @@ function isValidCode( serviceName = '', content = '' ) {
 	return includes( content, serviceIds[ serviceName ] );
 }
 
+function jetpackVersionUnsupported( site ) {
+	// TODO: replace with actual supported Jetpack version when that is known
+	return get( site, 'jetpack', false ) &&
+		versionCompare( get( site, 'options.jetpack_version', '0' ), '4.3.1', '<' );
```
Comment (vindl) [inline client/my-sites/site-settings/seo-settings/form.jsx]: This feature will probably be released with 4.4.

File: client/my-sites/site-settings/seo-settings/form.jsx [hunk 3]
```diff
@@ -419,12 +441,27 @@ export const SeoForm = React.createClass( {
 					</Notice>
 				}
 
+				{ showAdvancedSeo && isJetpackUnsupported &&
```
Comment (vindl) [inline client/my-sites/site-settings/seo-settings/form.jsx]: Maybe we should add Jetpack version check to `showAdvancedSeo` in order to hide form sections for older versions.

File: client/my-sites/site-settings/seo-settings/form.jsx [hunk 4]
```diff
@@ -99,6 +104,12 @@ function isValidCode( serviceName = '', content = '' ) {
 	return includes( content, serviceIds[ serviceName ] );
 }
 
+function jetpackVersionUnsupported( site ) {
```
Comment (mtias) [inline client/my-sites/site-settings/seo-settings/form.jsx]: It seems we shouldn't have these kind of functions on the component. What about a selector?

**Comment by roundhill** [comment]
Added a Preview button in 5cbc186 that only shows for Jetpack sites:

https://cloudup.com/c1q18J4jZ5L

**Comment by mtias** [comment]
> Added a Preview button in 5cbc186 that only shows for Jetpack sites:

What does this one do?

**Comment by roundhill** [comment]
> What does this one do?

This shows the `SEO` preview tab in a `WebPreview` component for Jetpack sites. See: https://github.com/Automattic/wp-calypso/pull/8663#issuecomment-256194097

**Comment by mtias** [comment]
Ah, `defaultViewportDevice="seo"`.

**Comment by jeherve** [comment]
We'll ship the first 4.4 Beta soon, so it would be nice if this were available for Jetpack Beta testers, with the version check @roundhill mentioned above.

Maybe we could hide the current Editor SEO tools behind that version check at the same time. See #7020.

**Comment by roundhill** [comment]
> Maybe we could hide the current Editor SEO tools behind that version check at the same time.

1e80c05d586ac74fb9ee4f39e62f66f40018d14b essentially did this, if the module isn't active you won't see the accordion in the editor.

**Comment by roundhill** [comment]
> Should be fine now. We should double check that SEO tab and Editor accordion work properly for WordPress.com sites before deploying.

I tested this case this morning for sites with and without the business plan.

**Comment by rodrigoi** [comment]
Verified to work as expected for Jetpack and WordPress.com sites. All nudges, warnings and editor drawer items show correctly. Tested with and without upgrades, with the default Jetpack configuration.

**Pull request merged**